dining/views.py

```diff
@@ -16,6 +14,462 @@
+                    "days"
+                ][0]["cafes"][venue_id]["name"]
+            formatted_date = datetime.datetime.strptime(date, "%Y-%m-%d").strftime("%-m/%d/%Y")
+            days.append(
+                {"tblDayPart": get_meals(v2_response, venue_id), "menudate": formatted_date}
+            )
+        response["result_data"]["Document"]["tblMenu"] = days
+
+        return Response(normalize_weekly(response)["result_data"])
+
+
+class DailyMenu(APIView):
+    """
+    GET: Returns data on daily menu for a particular venue
+    """
+
+    def get(self, request, venue_id):
+        date = str(timezone.localtime().date())
+        try:
+            v2_response = dining_request(V2_ENDPOINTS["MENUS"] + venue_id + "&date=" + date)
+        except APIError as e:
+            return Response({"error": str(e.args[0])}, status=400)
+
+        response = {"result_data": {"Document": {}}}
+        response["result_data"]["Document"]["menudate"] = datetime.datetime.strptime(
+            date, "%Y-%m-%d"
+        ).strftime("%-m/%d/%Y")
+        if venue_id in VENUE_NAMES:
+            response["result_data"]["Document"]["location"] = VENUE_NAMES[venue_id]
+        else:
+            response["result_data"]["Document"]["location"] = v2_response["result_data"]["days"][0][
+                "cafes"
+            ][venue_id]["name"]
+        response["result_data"]["Document"]["tblMenu"] = {
+            "tblDayPart": get_meals(v2_response, venue_id)
+        }
+        return Response(response["result_data"])
+
+
+class DiningItem(APIView):
+    """
+    GET: Returns data on a particular item
+    """
+
+    def get(self, request, item_id):
+        try:
+            response = dining_request(V2_ENDPOINTS["ITEMS"] + item_id)
+        except APIError as e:
+            return Response({"error": str(e.args[0])}, status=400)
+        return Response(response["result_data"])
+
+
+class Preferences(APIView):
+    """
+    GET: returns list of a User's diningpreferences
+
+    POST: updates User dining preferences by clearing past preferences
+    and resetting them with request data
+    """
+
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request):
+
+        preferences = DiningPreference.objects.filter(profile=request.user.profile)
+
+        # aggregated venues and puts it in form {"venue_id": x, "count": x}
+        aggregated_preferences = preferences.values("venue_id").annotate(count=Count("venue"))
+
+        # switches venue's auto-generated id with Penn's venue_id
+        for preference in aggregated_preferences:
+            venue = Venue.objects.get(id=preference["venue_id"])
+            preference["venue_id"] = venue.venue_id
+
+        return Response({"preferences": aggregated_preferences})
+
+    def post(self, request):
+
+        profile = request.user.profile
+
+        # clears all previous preferences associated with the profile
+        preferences = DiningPreference.objects.filter(profile=profile)
```

Comment:
`DiningPreference.objects.filter(profile=profile).delete()` should do this in one line